src/main/java/appeng/container/implementations/WirelessCraftConfirmContainer.java

```diff
@@ -184,18 +185,19 @@ private boolean cpuMatches(final ICraftingCPU c) {
     }
 
     public void startJob() {
-        ScreenHandlerType<?> originalGui = null;
-
         final IActionHost ah = getActionHost();
-        if(ah instanceof WCTGuiObject) originalGui = WCTContainer.TYPE;
-        else if(ah instanceof WPTGuiObject) originalGui = WPTContainer.TYPE;
+        if (ah instanceof WTGuiObject) {
+            ScreenHandlerType<?> originalGui = ((WTGuiObject) ah).type;
 
-        if(result == null && isSimulation()) return;
+            if (result == null && isSimulation())
+                return;
 
-        setAutoStart(false);
-        if(((ICraftingGrid) getGrid().getCache(ICraftingGrid.class)).submitJob(result, null, selectedCpu, true, getActionSrc()) != null && originalGui != null && getLocator() != null) {
-            if(originalGui.equals(WCTContainer.TYPE)) WCTContainer.open(getPlayerInventory().player, getLocator());
-            else if(originalGui.equals(WPTContainer.TYPE)) WPTContainer.open(getPlayerInventory().player, getLocator());
+            setAutoStart(false);
+            if (((ICraftingGrid) getGrid().getCache(ICraftingGrid.class)).submitJob(result, null, selectedCpu, true, getActionSrc()) != null && getLocator() != null) {
+                if (originalGui.equals(WCTContainer.TYPE)) WCTContainer.open(getPlayerInventory().player, getLocator());
+                else if (originalGui.equals(WPTContainer.TYPE))
+                    WPTContainer.open(getPlayerInventory().player, getLocator());
+            }
```

Comment:
I am not entirely sure, but couldn't we just store a functional interface for W?Container.open(...)?